roles/rsyslog/tasks/main_core.yml

```
@@ -3,51 +3,82 @@
 
 # if inputs are given and rsyslog_enabled is true, rsyslog.conf will be
 # overwritten, so no need to reinstall package
-- name: Reinstall package to restore rsyslog config if purging
+- name: Check if package needs to get reinstalled to restore rsyslog config if purging
   when:
     - logging_purge_confs | bool | d(false)
-    - not (rsyslog_inputs and __rsyslog_enabled | bool)
+    - not rsyslog_inputs | bool
+    # - not __rsyslog_enabled | bool
     - not __logging_is_ostree | d(false)
   block:
     # it is assumed that the only packages providing config files that might
-    # be modified are the base packages - if this is not so, then additional
-    # packages will need to be added to this check
+    # be modified are the base packages (but not others like iproute);
+    # if this is not so, then additional packages will need to be added to this check
     - name: Get status of rsyslog packages
       command: rpm -V {{ item }}  # noqa command-instead-of-module
-      loop: "{{ __rsyslog_base_packages }}"
+      loop: "{{ __rsyslog_base_packages  | select('search', 'rsyslog') | list }}"
```

```suggestion
      loop: "{{ __rsyslog_base_packages | select('search', 'rsyslog') | list }}"
```
ansible-lint requires exactly 1 space